common/v2/config/networks/globalProvider.ts

```diff
@@ -0,0 +1,81 @@
+import { BaseProvider } from 'ethers/providers';
+import { PROVIDER_OPTIONS } from './providerOptions';
+import { Network } from 'v2/services/Network/types';
+
+export type NetworkKey = keyof typeof PROVIDER_OPTIONS;
+
+type FallbackProvider = ethers.providers.FallbackProvider;
+const FallbackProvider = ethers.providers.FallbackProvider;
+
+type TempProviders = { [K in NetworkKey]: BaseProvider[] };
+type FallbackProviders = { [K in NetworkKey]: FallbackProvider };
+
+// tslint:disable-next-line
+function entries<T, K extends keyof T>(obj: { [K in keyof T]: T[K] }): [K, T[K]][] {
+  return Object.entries(obj) as any;
+}
+
+function createFallBackProvidersFrom(config: typeof PROVIDER_OPTIONS): FallbackProviders {
+  const tempProviders: TempProviders = {} as TempProviders;
+
+  // create fallback providers
+  for (const [networkKey, providerUrls] of entries(config)) {
+    for (const url of providerUrls) {
+      if (!tempProviders[networkKey]) {
+        tempProviders[networkKey] = [];
+      }
+      if (url) {
+        if (url.includes('etherscan')) {
+          const network = url.split('+')[1];
+          tempProviders[networkKey].push(new ethers.providers.EtherscanProvider(network));
+        } else {
+          tempProviders[networkKey].push(new ethers.providers.JsonRpcProvider(url));
+        }
+      }
+    }
+  }
+
+  const fallBackProviders: FallbackProviders = {} as FallbackProviders;
+  for (const [networkKey, providers] of entries(tempProviders)) {
+    fallBackProviders[networkKey] = new ethers.providers.FallbackProvider(providers);
+  }
+  return fallBackProviders;
+}
+
+export const createProviderHandler = (network: Network): FallbackProvider => {
+  const newProviderPattern: any = { [network.name]: [] };
+  network.nodes.forEach(node => {
+    if (node.url && /^https?:\/\/.+/i.test(node.url)) {
+      // Not very-well covered test for if url is a valid url (sorts out web3 nodes / non-https nodes).
+      newProviderPattern[network.name].push(node.url);
+    }
+  });
+  return createFallBackProvidersFrom(newProviderPattern)[network.name as NetworkKey];
+};
+
+export const allProviders: FallbackProviders = createFallBackProvidersFrom(PROVIDER_OPTIONS);
+
+type FilterFlags<Base, Condition> = {
+  [Key in keyof Base]: Base[Key] extends Condition ? Key : never
+};
+type AllowedNames<Base, Condition> = FilterFlags<Base, Condition>[keyof Base];
+
+type SubType<Base, Condition> = Pick<Base, AllowedNames<Base, Condition>>;
+
+type ProviderMethod = SubType<FallbackProvider, (...args: any) => any>;
+
+async function callProviderMethod<K extends keyof ProviderMethod>(
+  method: K,
+  args: { [NetworkName in NetworkKey]?: Parameters<ProviderMethod[K]> }
+) {
+  const arrayOfResults = [];
+  for (const network of Object.keys(args)) {
+    const provider = allProviders[network as NetworkKey];
+    const argsForNetwork: any = args[network as NetworkKey];
+    arrayOfResults.push(await (provider[method] as any)(...argsForNetwork));
+  }
```

Comment:
I'm not sur I understand the use of `args` nor where its coming from.
Shouldn't it be the arguments for the method? 
Are they different per network?
If we only need the response from one network, why do we return multiple results?
I'm concerned that the await will block the thread too much.
Can you share a usage example and with arguments and return value?

Reply:
The methods of Ethers.js all take in a different argument or set of arguments. These arguments need to be applied to the function call and the network. so you would do something like `callProviderMethod('getBalance', {
  Ethereum: ['0xceFB24f90dE062Ee1DaB076516E41993EC5c7FA8'],
  Kovan: ['0xceFB24f90dE062Ee1DaB076516E41993EC5c7FA8', 'latest']
  Kovan: ['0xceFB24f90dE062Ee1DaB076516E41993EC5c7FA8', 'latest'],
  Ethereum: ['0xceFB24f90dE062Ee1DaB076516E41993EC5c7FA8']
});`